packages/wrangler/src/module-collection.ts

```diff
@@ -92,6 +102,73 @@ export default function makeModuleCollector(props: {
           modules.splice(0);
         });
 
+        // ~ start  legacy module specifier support ~
+
+        // This section detects usage of "legacy" 1.x style module specifiers
+        // and modifies them so they "work" in wrangler v2, but with a warning
+
+        const rulesMatchers = rules.flatMap((rule) => {
+          return rule.globs.map((glob) => {
+            const regex = globToRegExp(glob);
+            return {
+              regex,
+              rule,
+            };
+          });
+        });
+
+        build.onResolve(
+          {
+            filter: new RegExp(
+              [...props.wrangler1xlegacyModuleReferences.fileNames]
+                .map((fileName) => `^${fileName}$`)
+                .join("|")
+            ),
+          },
+          async (args: esbuild.OnResolveArgs) => {
+            // In the future, this will simply throw an error
+            console.warn(
+              `Deprecation warning: detected a legacy module import in "./${path.relative(
+                process.cwd(),
+                args.importer
+              )}". This will stop working in the future. Replace references to "${
+                args.path
+              }" with "./${args.path}";`
+            );
+
+            // take the file and massage it to a
+            // transportable/manageable format
+            const filePath = path.join(
+              props.wrangler1xlegacyModuleReferences.rootDirectory,
+              args.path
+            );
+            const fileContent = await readFile(filePath);
+            const fileHash = crypto
+              .createHash("sha1")
+              .update(fileContent)
+              .digest("hex");
+            const fileName = `./${fileHash}-${path.basename(args.path)}`;
+
+            const { rule } =
+              rulesMatchers.find(({ regex }) => regex.test(fileName)) || {};
+            if (rule) {
+              // add the module to the array
+              modules.push({
+                name: fileName,
+                content: fileContent,
+                type: RuleTypeToModuleType[rule.type],
+              });
+              return {
+                path: fileName, // change the reference to the changed module
+                external: props.format === "modules", // mark it as external in the bundle
+                namespace: `wrangler-module-${rule.type}`, // just a tag, this isn't strictly necessary
+                watchFiles: [filePath], // we also add the file to esbuild's watch list
+              };
```

Comment:
There is a lot of duplication here with the actual module-collection plugin handlers.
I appreciate that you are trying to keep this bit separate so that it is easy to remove later.
But I think there is still room for creating some helper function or two to remove that duplication.

Reply:
honestly, it doesn't seem worth it. there's just enough differences interspersed in the body where the abstraction will be a weird argument signature, and it'll only get removed by 2.1. I'll give it a go when I try to implement bundle splitting, that might tease it apart some more.

Reply:
Cool. Thanks for taking a look.